### PR TITLE
Amplify installation fixes

### PR DESCRIPTION
**Pull request by cblanc**
Hit some issues while using this ansible role to setup the amplify agent on bionic

After applying the patches in this PR, appears to install without issue on my system

### CODE REVIEW

File: tasks/main.yml [hunk 1]
```diff
@@ -58,7 +58,6 @@
   when:
     - nginx_amplify_enable
     - nginx_amplify_api_key is defined
-    - nginx_amplify_api_key
```
Comment (alessfg) [inline tasks/main.yml]: Can you re-add this condition? It covers some edge cases where `nginx_amplify_api_key` is defined but is set as an empty string.

Comment (cblanc) [reply tasks/main.yml]: Sure. Just to be clear, returning the conditional restores the error:

```
fatal: [test]: FAILED! => {"msg": "The conditional check 'nginx_amplify_api_key' failed. The error was: template error while templating string: expected token 'end of statement block', got 'my_api_key_minus_first_2_chars'. String: {% if my_api_key %} True {% else %} False {% endif %}\n\n
```

Not sure whats going on here but I found one or two open issues where ansible doesn't evaluate conditionals on variables correctly in some scenarios. I didn't spend too much time diving into it however...

I found a solution [here](https://github.com/ansible/ansible-lint/issues/457) which favours `when: my_var | length > 0` - the downside being it's not idiomatic python (so I'm told)

The alternative is `when: my_var != ""` but apparently this clashes with linting rule 602

Comment (alessfg) [reply tasks/main.yml]: Oh interesting. Last I tested with Ansible 2.4 installing Amplify with my API Key worked just fine. For now given the approach you used looks like the best solution though. We'll see if Ansible 2.8 fixes this bug.
